### PR TITLE
Cerberus owners in config, create pdv command updated

### PR DESCRIPTION
**Pull request by megaherz**

### CODE REVIEW

File: x/pdv/handler.go [hunk 1]
```diff
@@ -16,47 +12,33 @@ import (
...
-
-func handleMsgCreatePDV(ctx sdk.Context, cerberus cerberusapi.Cerberus, keeper Keeper, tokensKeeper token.Keeper, msg MsgCreatePDV) (*sdk.Result, error) {
-	meta, err := cerberus.GetPDVMeta(ctx.Context(), msg.Owner.String(), msg.ID)
-	if err != nil {
-		if errors.Is(err, cerberusapi.ErrNotFound) {
-			ctx.Logger().Error("pdv doesn't exist", "owner", msg.Owner, "id", msg.ID)
-			return nil, errors.New("pdv does not exist")
+func handleMsgCreatePDV(ctx sdk.Context, keeper Keeper, tokensKeeper token.Keeper, msg MsgCreatePDV) (*sdk.Result, error) {
+	cerberuses := keeper.GetCerberuses(ctx)
+	var isCerberus bool
```
Comment (mikluke) [inline x/pdv/handler.go:29]:
```
for _, v := range cerberusOwners {
  addr, _ := ssdk.AccAddressFromBech32(v)
  if msg.Owner.Equals(addr) && !addr.Empty() {
  	tokensKeeper.AddTokens(ctx, msg.Receiver, sdk.NewIntFromUint64(msg.Reward), utils.GetHash(msg))
  	return &sdk.Result{}, nil
  }
}

return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Owner is not Cerberus")
```

looks better, I think

File: x/pdv/types/msg.go [hunk 2]
```diff
@@ -7,15 +7,17 @@ import (
 
 // MsgCreatePDV defines a CreatePDV message
 type MsgCreatePDV struct {
-	Owner sdk.AccAddress `json:"owner"`
-	ID    uint64         `json:"id"`
+	Owner    sdk.AccAddress `json:"owner"`
+	Receiver sdk.AccAddress `json:"receiver"`
+	Reward   uint64         `json:"reward"`
```
Comment (mikluke) [inline x/pdv/types/msg.go:12]: Looks like we shouldn't remove ID field

This is a ledger and I think it's fair to keep a link to pdv there

**Review by mikluke — CHANGES_REQUESTED** [review]

File: x/pdv/genesis.go [hunk 3]
```diff
@@ -1,40 +1,53 @@
...
+	"fmt"
+	"github.com/Decentr-net/decentr/x/pdv/types"
+	"github.com/cosmos/cosmos-sdk/codec"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 type GenesisState struct {
-	CerberusAddr string `json:"cerberus_address"`
+	Cerberuses []string `json:"cerberuses"`
```
[the remark this answers is not in the document]
Comment (megaherz) [reply x/pdv/genesis.go]: Sounds better, renamed

File: x/pdv/genesis.go [hunk 4]
```diff
@@ -1,40 +1,53 @@
...
+	if appState[ModuleName] != nil {
+		cdc.MustUnmarshalJSON(appState[ModuleName], &genesisState)
 	}
+
+	return genesisState
 }
 
 // DefaultGenesisState returns a default genesis state
 func DefaultGenesisState() GenesisState {
-	return NewGenesisState("https://cerberus.testnet.decentr.xyz")
```
[the remark this answers is not in the document]
Comment (megaherz) [reply x/pdv/genesis.go:22]: Moved to the config
```json
{
  "network": {
    "chainId": "testnet",
    "rest": [
      "https://ares.testnet.decentr.xyz",
      "https://poseidon.testnet.decentr.xyz",
      "https://hera.testnet.decentr.xyz",
      "https://zeus.testnet.decentr.xyz",
      "https://hermes.testnet.decentr.xyz"
    ]
  },
  "minVersion": "1.1.2",
  "cerberus": {
    "url": "https://cerberus.testnet.decentr.xyz",
    "minPDVCount": 50,
    "maxPDVCount": 100,
    "rewards": {
      "cookie": 2,
      "login_cookie": 4
    }
  }
}
```
It is going to be testnet2. No need to have a backward compatibility.

[the remark this answers is not in the document]
Comment (megaherz) [reply x/pdv/genesis.go:22]: Already updated at https://tf-decentr-public-testnet.s3.us-east-2.amazonaws.com/config.json

File: x/pdv/types/msg.go (hunk 2 above)
Comment (megaherz) [reply x/pdv/types/msg.go:13]: Indeed, probably we will add some cli/rest to query them

File: x/pdv/handler.go (hunk 1 above)
Comment (megaherz) [reply x/pdv/handler.go]: 👍🏻 refactored

**Pull request merged**